### ISSUE TITLE
InputText.value is not set when an empty String is used

### ISSUE THREAD

**Issue by juancastillo0** [erlage/rad#13]
Hi, thanks for your work!

I found that when I pass an empty String to the `InputText` widget constructor, the text is not cleared. The value of the element is not overridden. Should I save the `dart:html.InputElement` reference to clear and update the value, like a Flutter TextEditingController? 

I was also wondering whether a `void Function(String value) onValueChange` argument in the constructor could be used as API, maybe the same for others where we know the Dart type such as number or date inputs. At the moment I was casting the event target to `dart:html.InputElement` and accessing the value from that. However, for `InputText` and others we know the element and type of input (and Dart's value type).

The full code that I was testing is in [this file](https://github.com/juancastillo0/room_signals/blob/61fad9905df96129816ddb31e4691c1ea0465657/web_client/web/src/room_client_view.dart#L293).

Thanks again!

**Comment by erlage** [comment]
Hi, thanks for trying it out.

> found that when I pass an empty String to the InputText widget constructor, the text is not cleared. 

Just checked, it's indeed a bug, thanks for reporting it.

> I was also wondering whether a void Function(String value) onValueChange argument in the constructor could be used as API, maybe the same for others where we know the Dart type such as number or date inputs. 

Do you mean adding `onValueChange` like this?

```dart
InputText(
  onValueChange: (value) => print('updated value is $value'), 
)
```

**Comment by juancastillo0** [comment]
Thanks for the response.

> Do you mean adding `onValueChange` like this?

Yes, similar to Flutter's [`TextField.onChanged`](https://api.flutter.dev/flutter/material/TextField/onChanged.html) argument. Maybe for other Input types could be a specific Dart type like double for [`InputNumber`](https://pub.dev/documentation/rad/latest/widgets_html/InputNumber-class.html).

```dart
InputNumber(
  onValueChange: (double? value) => print('updated value is $value'), 
)
```

The value is nullable when it could not be parsed or is empty? or pass the raw String value and the parsed value as arguments to the `onChanged` callback. Or maybe a class with most of the event's info:

```dart
class InputValueChanged<T> {
  final T value;
  final EmittedEvent event;

  html.InputElement get element => event.target as html.InputElement;
  String get rawValue => element.value;
}
```

But that is probably too complicated and would be strange for `InputText` since `value` would be the same as `rawValue`. Maybe just the:

```dart
InputNumber(
  onValueChange: (String value) => print('updated value is $value'), 
)
```
would be nice enough, since one does not need to extract the target from the event and cast it into `html.InputElement`.

**Comment by erlage** [comment]
I too think a basic `onValueChange(String value)` would be great. I'll give it a try.

Talking about issue, I've published a fix in [`v1.0.2`](https://pub.dev/packages/rad/versions/1.0.2), let me know if that works.

**Comment by juancastillo0** [comment]
Yes, It's working perfectly now. Thank you for your help!
